### ISSUE TITLE
PVs in same context with different "forms" may interfere with one another

### ISSUE THREAD

**Issue by klauer** [pyepics/pyepics#159]
Background:
* `_PVcache_` is keyed on `(pvname, form, context)`
* `epics.ca` cache is (and has always been) keyed on `epics.ca._cache[context][pvname]`
* This means that two `epics.PV`s can have a reference to the same `chid` (i.e., channel id from libca)

The above becomes an issue because:
* `PV.disconnect()` is called on `PV.__del__`; and
* `PV.disconnect()` removes the CA cache item, meaning other `epics.PV`s of different forms (which refer to the same chid, and that same CA cache entry) may break

**Comment by newville** [comment]
@klauer Yes, good points.   Ideally, the user didn't even have to know that there were different forms, but maybe that is unrealistic.  

I guess I don't have a strong opinion on whether `PV.disconnect()` *should* delete all cache entries for that channel. I probably sort of lean toward "let most channels stay connected for as long as possible", but it's not a strong opinion.

Anyway, it would be fine with me to have the CA cache be further keyed by "form".

**Comment by klauer** [comment]
I think it may not be worth fixing at the moment if no one is getting bit by it. At least we can keep it in the back of our mind when trying to diagnose issues.

**Comment by newville** [comment]
@klauer OK, that sounds fine -- it should be solved but appears to be non-urgent.
